cmd/ctl/check_config.go

```
@@ -0,0 +1,580 @@
+		cm, err := kc.Clientset.CoreV1().ConfigMaps(ns).Get(ctx, "argocd-cm", metav1.GetOptions{})
+		if err != nil {
+			// If both CR and ConfigMap are not found, return error
+			if k8serrors.IsNotFound(err) && crNotFound {
+				return fmt.Errorf("neither ArgoCD CR nor argocd-cm ConfigMap found in namespace %s", ns)
+			}
+			// Return all other errors
+			return fmt.Errorf("failed to get argocd-cm ConfigMap: %w", err)
+		}
+		if v := strings.TrimSpace(cm.Data["application.namespaces"]); v != "" {
+			appNamespacesSet = true
+		}
+	}
+
+	if appNamespacesSet {
+		return fmt.Errorf("argo CD configured for namespaced mode (applicationNamespaces is set), must be configured for cluster-scoped mode")
+	}
+
+	// Verify cluster-scoped behavior by checking if Applications can be accessed across namespaces
+	// In cluster-scoped mode, we should be able to list Applications from any namespace (even though
+	// Applications themselves are namespace-scoped resources). This verifies that Argo CD can operate
+	// in cluster-scoped mode, managing Applications across all namespaces.
+	appGVR := schema.GroupVersionResource{Group: "argoproj.io", Version: "v1alpha1", Resource: "applications"}
+
+	// List Applications across all namespaces (cluster-scoped operation)
+	// This is equivalent to: kubectl get applications --all-namespaces
+	_, err = kc.DynamicClient.Resource(appGVR).Namespace(metav1.NamespaceAll).List(ctx, metav1.ListOptions{Limit: 1})
+	if err != nil {
+		// If we can't list Applications cluster-wide, return the error
+		// This could indicate namespaced mode is restricting access or a permissions issue
+		return fmt.Errorf("failed to list Applications cluster-wide (required for cluster-scoped mode): %w", err)
+	}
+	// If list succeeds, we can confirm cluster-scoped mode is working
+	return nil
+}
+
+// routeAPIExists checks if the OpenShift Route API exists on the cluster.
+func routeAPIExists(kc *kube.KubernetesClient) bool {
+	if kc.Clientset == nil {
+		return false
+	}
```

Should return an error if `kc.Clientset` is nil